### PR TITLE
Add instrumentation tests for the GitHub Repository Settings Dialog

### PR DESCRIPTION
**Pull request by EmmanuelAB**

### CODE REVIEW

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package net.osmtracker.layouts;
+
+import android.support.test.espresso.ViewAssertion;
+import android.support.test.rule.ActivityTestRule;
+import android.view.View;
```
Comment (jamescr) [inline app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:5]: This import is not used.

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import net.osmtracker.OSMTracker;
+import net.osmtracker.R;
+import net.osmtracker.activity.AvailableLayouts;
+import net.osmtracker.util.TestUtils;
```
Comment (jamescr) [inline app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:10]: This import is not used. Delete it.

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        checkTextFieldsState(not(isEnabled()));
+        checkTextFieldsDefaultValues();
+
+        onView(withId(R.id.custom_server)).perform(click());
+        checkStateAfterToggle(R.id.custom_server, R.id.default_server);
+        checkTextFieldsState(isEnabled());
+    }
+
+    @Test
+    public void testRepositoryValidation(){
```
Comment (jamescr) [inline app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:51]: Isn't better to use the constant values for github username, repository name and branch name, declared in OSMTracker.Preferences. (Except in line 53 where the "NONE" in requiered).

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+     * Check if the dialog is shown by looking for its title on the screen
+     */
+    private void checkDialogState(ViewAssertion assertion) {
+        onView(withText(getStringResource(R.string.prefs_ui_github_repository_settings))).check(assertion);
+    }
+
+    /**
+     * Check that the text fields values match the expected default ones
+     */
+    private void checkTextFieldsDefaultValues() {
```
Comment (jamescr) [inline app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:95]: The EditText elements for github_username, repository_name and branch_name are always displayed, no matter if the custom repo or default server are selected. The property that changes when the server also changes is if they are enabled of not. So using `isDisplay()` in `check(matches()` seems not useful, I suggest to use `isEnable()`.

Also this function could received a param `Matcher matcher`, and a new call could be added after line 47 with a `isEnabled()` param. (In a similar way its done with `checkTextFieldsState()`

**Comment by jamescr** [comment]
Any idea on why Travis fails?

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java (hunk 4 above)
Comment (EmmanuelAB) [reply app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:95]: Oh ok, but those lines are checking that the OSMTracker.Preferences.VAL_X  strings are shown in the UI, not checking that the EditTexts themselves are enabled. 

For example when it says `onView(withText(OSMTracker.Preferences.VAL_GITHUB_USERNAME)).` it's checking that a view containing VAL_GITHUB_USERNAME is displayed and that only happens if the EditText value is such.

**Comment by EmmanuelAB** [comment]
> Any idea on why Travis fails?

Yes, It's already solved. It was because of the small screen of the emulator where the app tests run in Travis. When filling the fields of the dialog it was needed to close de keyboard for finding the "Save" button.

File: app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+        onView(withId(R.id.github_username)).perform(clearText(), typeText(user));
+        onView(withId(R.id.repository_name)).perform(clearText(), typeText(repo));
+        onView(withId(R.id.branch_name)).perform(clearText(), typeText(branch));
+
+        // This is required to find the "Save" button that will be clicked next, that button may be
+        // invisible because it's covered by the keyboard on small screens
+        Espresso.closeSoftKeyboard();
+
+        onView(withText(getStringResource(R.string.menu_save).toUpperCase())).perform(click());
```
Comment (jamescr) [inline app/src/androidTest/java/net/osmtracker/layouts/RepositorySettingsDialogTest.java:81]: Eliminating `toUpperCase()` [fixes the Travis error.](https://travis-ci.org/github/jamescr/osmtracker-android/builds/708953501)

**Pull request merged**